packages/fxa-auth-server/lib/routes/totp.js

```diff
@@ -268,6 +232,45 @@ module.exports = (
           tokenId: sessionToken && sessionToken.id,
         });
 
+        // Clean up the recovery phone if it was registered.
+        // Don't fail if this doesn't work, but monitor success rate with stats.
+        try {
+          const success = await recoveryPhoneService.removePhoneNumber(uid);
+          if (success) {
+            statsd.increment('totp.destroy.remove_phone_number.success');
+            await glean.twoStepAuthPhoneRemove.success(request);
+          } else {
+            statsd.increment('totp.destroy.remove_phone_number.fail');
+          }
+        } catch (error) {
+          if (
+            error instanceof RecoveryNumberNotExistsError ||
+            error instanceof RecoveryNumberRemoveMissingBackupCodes
+          ) {
+            statsd.increment('totp.destroy.remove_phone_number.fail');
+          } else {
+            statsd.increment('totp.destroy.remove_phone_number.error');
+            log.error('totp.destroy.remove_phone_number.error', error);
+          }
+        }
+
+        // Clean up any associated backup codes.
+        // Again, don't fail if errors out, but monitor with stats.
+        try {
+          const success = await backupCodeManager.deleteRecoveryCodes(uid);
+          if (success) {
+            statsd.increment('totp.destroy.delete_recovery_codes.success');
+          } else {
+            statsd.increment('totp.destroy.delete_recovery_codes.fail');
+          }
+        } catch (error) {
+          statsd.increment('totp.destroy.delete_recovery_codes.error');
+          log.error('totp.destroy.delete_recovery_codes.error', error);
+        }
+
+        // Record that the 2fa was successfully removed
+        glean.twoStepAuthRemove.success();
```

Comment:
Side note, do we have security events for the recovery phone endpoints? Might need to file a follow up if we dont.